qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -465,14 +465,95 @@ def __init__(self, name: str, instrument: B1517A, **kwargs: Any):
         self.source_voltage = _FMTResponse(None, None, None, None)
         self._fudge: float = 1.5
 
-    def get_raw(self) -> Tuple[Tuple[float, ...], Tuple[float, ...]]:
+    def set_names_labels_and_units(self,
+                                   names: Optional[Sequence[str]] = None,
+                                   labels: Optional[Sequence[str]] = None,
+                                   units: Optional[Sequence[str]] = None
+                                   ) -> None:
         measurement_mode = self.instrument.get_measurement_mode()
-        if len(measurement_mode['channels']) != 2:
-            raise ValueError('Two measurement channels are needed, one for '
-                             'gate current and other for source drain '
-                             'current.')
+        channels = measurement_mode['channels']
 
-        smu = self.instrument.by_channel[measurement_mode['channels'][0]]
+        if names is None:
+            names = [f"param{n+1}" for n in range(len(channels))]
```

Comment:
should this also be param{n+1}_current to match the label?

Reply:
that would be backwards incompatible :(